crates/sui-core/src/authority_active/execution_driver/mod.rs

```diff
@@ -1,46 +1,49 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-use std::{collections::HashSet, sync::Arc};
-use sui_types::{base_types::TransactionDigest, error::SuiResult, messages::VerifiedCertificate};
-use tracing::{debug, error, info};
-
-use crate::authority::{AuthorityState, PendingDigest};
-use crate::authority_client::AuthorityAPI;
-
-use futures::{stream, StreamExt};
-
 use super::ActiveAuthority;
-
-use tap::TapFallible;
+use crate::authority::AuthorityState;
+use crate::authority_client::AuthorityAPI;
+use async_trait::async_trait;
+use std::{sync::Arc, time::Duration};
+use sui_metrics::spawn_monitored_task;
+use sui_types::{
+    error::{SuiError, SuiResult},
+    messages::VerifiedCertificate,
+};
+use tokio::{
+    sync::Semaphore,
+    time::{sleep, timeout},
+};
+use tracing::{debug, error, info, warn};
 
 #[cfg(test)]
 pub(crate) mod tests;
 
+// Maximum number of certificates to be executed concurrently.
+// TODO: make this configurable.
+const MAX_EXECUTION_CONCURRENCY: usize = 20;
+const MAX_EXECUTION_TIMEOUT: Duration = Duration::from_secs(30);
```

Comment:
I don't believe we need a timeout here. We certainly have never observed a case of certificate execution hanging, and if we did it would be a massive bug that we would need to understand and fix. In addition, there are very few true `await` points in the execution path (most of them are compiled out) so a timeout can't even help very much.

The only place I can think of where a long wait is somewhat plausible is when we call `acquire_tx_guard()` - I say `somewhat` because any holder of the tx lock should finish quickly, therefore we should never wait very long. But if we were truly worried about that, we should put the timeout around `acquire_tx_guard()`, not here.

Reply:
I took the timeout behavior from node sync, but I agree probably we don't need it here. Handling a certificate that timed out is really strange too. Removed the timeout for now.